modules/cas_cache/layer_cache_management.c

```diff
@@ -2270,12 +2296,35 @@ static int cache_mngt_check_bdev(struct ocf_mngt_cache_device_config *cfg,
 
 	is_part = (cas_bdev_whole(bdev) != bdev);
 	part_count = cas_blk_get_part_count(bdev);
+
+	if (reattach) {
+		ENV_BUG_ON(!cache);
+
+		cache_priv = ocf_cache_get_priv(cache);
+		tmp_limits = cache_priv->device_properties.queue_limits;
+
+		reattach_properties_diff = blk_stack_limits(&tmp_limits,
+				&bdev->bd_disk->queue->limits, 0);
+		reattach_properties_diff |= tmp_limits.misaligned;
+		reattach_properties_diff |= (
+				CAS_CHECK_QUEUE_FLUSH(bdev->bd_disk->queue) !=
+				cache_priv->device_properties.flush
+				);
+		reattach_properties_diff |= (
+				CAS_CHECK_QUEUE_FUA(bdev->bd_disk->queue) !=
+				cache_priv->device_properties.fua
+				);
```

Comment:
I think the user would appreciate an information which particular properties doesn't match. How about extending these checks with logging?